jsonb-generator/src/main/java/io/avaje/jsonb/generator/ProcessingContext.java

```diff
@@ -57,16 +59,25 @@ static void addImportedPrism(ImportPrism prism, Element element) {
     final var json = CTX.get().importedJsonMap;
     final var subtypes = CTX.get().importedSubtypeMap;
     prism.value().forEach(m -> {
+      addImportedType(m);
```

Comment:
I think I need to review how this works for cascaded types referenced via an imported type. Looks like they will be treated as not imported.

Reply:
well, it seems you got me again, I'll need to add a module check to see if the class is part of the same module

Reply:
Other options are:
- Not cascade from imported types (so must be explicit instead)
- Change when it cascades from an imported type mark that type also as imported
- Let it happen as is ... and then people explicitly add imported types until it works

Reply:
possibly, but using modules to solve it is pretty easy. If it's not part of the module being compiled, it should get a `.json` package to avoid a split package situation.